data/physics_based_pi_formatter.py

```
@@ -0,0 +1,800 @@
+    from systems.primary.reactor.physics.thermal_hydraulics import ThermalHydraulicsModel
+    from systems.secondary.water_chemistry import WaterChemistry, WaterChemistryConfig
+    from systems.secondary.feedwater.level_control import ThreeElementControl, ThreeElementConfig
+    from systems.secondary.turbine.enhanced_physics import EnhancedTurbinePhysics, EnhancedTurbineConfig
+    from systems.secondary.steam_generator.enhanced_physics import EnhancedSteamGeneratorPhysics, EnhancedSteamGeneratorConfig
+    from systems.secondary.turbine.rotor_dynamics import RotorDynamicsModel, RotorDynamicsConfig
+    from systems.secondary.steam_generator.tsp_fouling_model import TSPFoulingModel
+    PHYSICS_MODULES_AVAILABLE = True
+except ImportError:
+    # Fallback for standalone operation
+    PHYSICS_MODULES_AVAILABLE = False
+    warnings.warn("Physics modules not available - using simplified validation")
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)